core/common/src/ContourDisplay.ts

```diff
@@ -0,0 +1,263 @@
+        return diff;
+      if ((diff = compareNumbers(lhs.majorPixelWidth, rhs.majorPixelWidth)) !== 0)
+        return diff;
+      if ((diff = compareNumbers(lhs.minorPixelWidth, rhs.minorPixelWidth)) !== 0)
+        return diff;
+      if ((diff = compareNumbers(lhs.majorPattern, rhs.majorPattern)) !== 0)
+        return diff;
+      if ((diff = compareNumbers(lhs.minorPattern, rhs.minorPattern)) !== 0)
+        return diff;
+      if ((diff = compareNumbers(lhs.minorInterval, rhs.minorInterval)) !== 0)
+        return diff;
+      if ((diff = compareNumbers(lhs.majorIntervalCount, rhs.majorIntervalCount)) !== 0)
+        return diff;
+
+      return diff;
+    }
+
+    private constructor(json?: ContourProps) {
+      if (undefined === json) {
+        this.majorColor = ColorDef.black;
+        this.minorColor = ColorDef.black;
+        this.majorPixelWidth = 2;
+        this.minorPixelWidth = 1;
+        this.majorPattern = 0;
+        this.minorPattern = 0;
+        this.minorInterval = 1;
+        this.majorIntervalCount = 5;
+      } else {
+        this.majorColor = json.majorColor ? ColorDef.create(json.majorColor) : ColorDef.black;
+        this.minorColor = json.minorColor ? ColorDef.create(json.minorColor) : ColorDef.black;
+        this.majorPixelWidth = json.majorPixelWidth ?? 2;
+        this.minorPixelWidth = json.minorPixelWidth ?? 1;
+        this.majorPattern = json.majorPattern ?? 0;
+        this.minorPattern = json.minorPattern ?? 0;
+        this.minorInterval = json.minorInterval ?? 1;
+        this.majorIntervalCount = json.majorIntervalCount ?? 5;
+      }
+    }
+
+    public static fromJSON(json?: ContourProps) {
+      return json ? new Contour(json) : new Contour({});
+    }
+
+    public toJSON(): ContourProps {
+      const props: ContourProps = {};
+
+      if (!this.majorColor.equals(ColorDef.black))
+        props.majorColor = this.majorColor.toJSON();
+
+      if (!this.minorColor.equals(ColorDef.black))
+        props.minorColor = this.minorColor.toJSON();
+
+      if (2 !== this.majorPixelWidth)
+        props.majorPixelWidth = this.majorPixelWidth;
+
+      if (1 !== this.minorPixelWidth)
+        props.minorPixelWidth = this.minorPixelWidth;
+
+      if (0 !== this.majorPattern)
+        props.majorPattern = this.majorPattern;
+
+      if (0 !== this.minorPattern)
+        props.minorPattern = this.minorPattern;
+
+      if (1 !== this.minorInterval)
+        props.minorInterval = this.minorInterval;
+
+      if (5 !== this.majorIntervalCount)
+        props.majorIntervalCount = this.majorIntervalCount;
+
+      return props;
+    }
+  }
+
+  /** JSON representation of the description of how contours should appear for a particular set of subcategories. */
+  export interface TerrainProps {
+    /** See [[Terrain.contourDef]]. */
+    contourDef?: ContourProps;
+    /** See [[Terrain.subCategories]]. */
+    subCategories?: Id64String[];
```

Comment:
Use `CompressedId64Set`.

Reply:
This has been done in the latest code push.